### PR TITLE
README.md #11

### PR DESCRIPTION
**Pull request by gentilboulet**
Madness

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,6 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+# vim tmp
```
Comment (CyrilleHugues) [inline .gitignore]: you so ninja

File: README.md [hunk 2]
```diff
@@ -3,7 +3,78 @@ Legends of the Wulin Character Sheet
 
 <p align="center">
     <a href="https://circleci.com/docs/1.0/status-badges/">
-        <img src="https://circleci.com/gh/gentilboulet/LotWCS.svg?&style=shield&circle-token=784ac32effdcc6931968ae398ce0c87289181e68"
+        <img src="https://circleci.com/gh/gentilboulet/LotWCS.svg?&style=shield"
```
Comment (CyrilleHugues) [inline README.md]: why ?

File: README.md [hunk 3]
```diff
@@ -3,7 +3,78 @@ Legends of the Wulin Character Sheet
...
+React Components are subdivided by responsibility: container Components are there to link the store to render Components.
+
+## File hierarchy
+The src directory is divided by responsibility:
+
+### actions
+Here are the action creators, a lot of helper functions present to generate an input to a specific action understandable by the reducers.
+
+### components
+__This directory regroups "renderer" components, handling only rendering and catching user inputs. __
```
Comment (CyrilleHugues) [inline README.md]: you have a [strange behavior](https://github.com/gentilboulet/LotWCS/blob/9320442318ee29f68484a6059692ebd00b1c5810/README.md) on github

File: README.md [hunk 4]
```diff
@@ -3,7 +3,78 @@ Legends of the Wulin Character Sheet
...
+
+### constants
+Constants definitions, mainly used to specify available action.type value, on which the reducers rely to identify actions.
+
+### containers
+__This directory regroups "containers" components and other functions for reading elements from the state. __
+
+Those are obtained by connecting the store and the state to a "renderer" component.
+To do so, three steps are taken for each one:
+* mapStateToProps: maps the state to the props, by transforming it, by adding information from static data ;
```
Comment (CyrilleHugues) [inline README.md]: refer to the [official documentation](https://github.com/reactjs/react-redux/blob/9961062afe35a9147b0ea819e7087170e66cd5f5/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options)

File: README.md (hunk 2 above)
Comment (gentilboulet) [inline README.md]: The token seems not needed to show the shield.

File: README.md (hunk 3 above)
Comment (gentilboulet) [inline README.md]: I came from the space before __.

File: README.md [hunk 5]
```diff
@@ -3,7 +3,77 @@ Legends of the Wulin Character Sheet
...
+Here are the global reducer, the one plugged to the Redux Store and a lot of sub reducers, handling a specific subset of actions.
+One important rule to note is that no action is applied to the state by different reducers.
+
+### types
+Typescript types definitions.
+
+## Data flow
+The data trickle down from the state to feed the props of the components.
+
+When a container maps the state to the props of a component, it must check which actions are available or not, and it must take prepare the cost of the action by checking for discount.
```
Comment (CyrilleHugues) [inline README.md]: must take prepare ?

Comment (CyrilleHugues) [inline README.md]: It's not really clear you're writing about skills costs and not js performance cost of the action.
You might want to write a short description before to explain why cost management is such a pain and here explaining your technical strategy as you will explain it in your next reactJs/humanTalks meetup :)

**Pull request merged**